### PR TITLE
Add test for arcs_manifest rule

### PR DESCRIPTION
**Pull request by csilvestrini**
Test checks that all the important files are included in the filegroup.
i.e. other imported .arcs files, particle implementations in .wasm/.js
files, .json data files.

### CODE REVIEW

File: src/tools/manifest-checker.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+//
+// To be run with the arcs_manifest BUILD rule.
+
+/**
+ * Loads the given .arcs manifest file and checks it for errors. Errors are
+ * thrown as an exception.
+ */
+async function checkManifest(src: string) {
+  const loader = new PlatformLoader({});
+  const manifest = await Manifest.load(src, loader);
```
Comment (cromwellian) [inline src/tools/manifest-checker.ts]: I'd verify that relative import statements are working. In the past we ran into some issues (especially in g3) with the loader not handling relative imports. They seem to have gone away, but I'd at least try it manually locally to make sure.

File: src/tools/manifest-checker.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      .map(error => error.toString());
+
+  if (manifestErrors.length) {
+    throw manifestErrors.join('\n');
+  }
+
+  // Check particle impls can be loaded.
+  for (const particle of manifest.particles) {
+    const implFile = particle.implFile;
+    if (implFile.endsWith('.wasm')) {
```
Comment (cromwellian) [inline src/tools/manifest-checker.ts]: Is this backwards? Should that be if !implFile.endsWith('.wasm') ?

File: src/tools/manifest-checker.ts (hunk 1 above)
Comment (csilvestrini) [reply src/tools/manifest-checker.ts]: Yep they work. I was testing the file `particles/Native/Wasm/example.arcs`, and played around with all the interesting cases, like relative imports, stores, etc.

I considered writing unit tests for this manifest-checker.ts file, but ultimately decided it wasn't worth it. I'd have to override too much of the Loader class to get the unit tests to work in a fake environment, and at that point we aren't really testing the implementation anymore.

File: src/tools/manifest-checker.ts (hunk 2 above)
Comment (csilvestrini) [reply src/tools/manifest-checker.ts]: Good catch, thanks

**Pull request merged**